lib/extensions.js

```diff
@@ -73,29 +92,31 @@ module.exports = function (realmConstructor) {
     enumerable: false,
   });
 
-  const getInternalCacheId = (realmObj) => {
-    const { name, primaryKey } = realmObj.objectSchema();
-    const id = primaryKey ? realmObj[primaryKey] : realmObj._objectKey();
-    return `${name}#${id}`;
-  };
+  Object.defineProperty(realmConstructor.Dictionary.prototype, "toJSON", {
+    value: function toJSON(_, cache = new Map()) {
+      const result = {};
+      for (const k of this._keys()) {
+        const v = this.getter(k);
```

Comment:
this does not seem to be the cleanest approach to iterate through a dictionary keys/values... but I had a lot of trouble trying other ways.

Reply:
For what it's worth, it seem equally bad as what you're removing from dictionary.js.